sphinx_markdown_builder/translator.py

```diff
@@ -577,11 +579,26 @@ def visit_desc_signature(self, node):
             for anchor in node.get("ids", []):
                 self._add_anchor(anchor)
 
-        # We don't want methods to be at the same level as classes,
-        # If signature has a non-null class, that's means it is a signature
-        # of a class method
-        h_level = 4 if node.get("class", None) else 3
-        self._push_context(TitleContext(h_level))
+        if (parent := node.parent) and "attribute" in parent.get("classes", ""):
+            self.add("- ", prefix_eol=1)
+        else:
+            # We don't want methods to be at the same level as classes,
+            # If signature has a non-null class, that's means it is a signature
+            # of a class method
+            h_level = 4 if node.get("class", None) else 3
+            self._push_context(TitleContext(h_level))
+
+    visit_desc_name = visit_literal
+    depart_desc_name = depart_literal
+
+    @pushing_context
+    def visit_desc_annotation(self, node):
+        self.add(" ")
```

Comment:
This solution is very specific to Python annotation. I'm not sure how this will affect other languages.
I've already implemented a workaround for this issue. It is in the main branch.